### PR TITLE
fix issue around formState with reset

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -234,6 +234,7 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
       getFieldValue(fieldsRef.current, fieldsRef.current[name]!.ref);
 
     if (isFieldArray) {
+      console.log(defaultValuesRef.current);
```
Comment (kotarella1110) [inline src/useForm.ts]: this line is unncessary.

File: src/useForm.ts [hunk 2]
```diff
@@ -319,9 +320,8 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
 
       if (isArray(payload)) {
         payload.forEach(name => {
-          const error = get(errors, name);
-          if (error) {
-            set(errorsRef.current, name, error);
+          if (errors[name]) {
+            set(errorsRef.current, name, errors[name]);
```
Comment (kotarella1110) [inline src/useForm.ts]: Why did you undo the fix for this commit (https://github.com/react-hook-form/react-hook-form/commit/065ab57af99eab29eab8164a5686d77c32430978)?

File: src/useForm.ts [hunk 3]
```diff
@@ -868,7 +868,7 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
 
   const handleSubmit = useCallback(
     (callback: OnSubmit<FormValues>) => async (
-      e?: React.BaseSyntheticEvent,
+      e: React.BaseSyntheticEvent,
```
Comment (kotarella1110) [inline src/useForm.ts]: Why did you undo the fix for this PR (https://github.com/react-hook-form/react-hook-form/pull/885)?

**Review by kotarella1110 — COMMENTED** [review]
@bluebill1049 please reply :pray:

File: src/useForm.ts (hunk 1 above)
Comment (bluebill1049) [reply src/useForm.ts]: Oops

File: src/useForm.ts (hunk 2 above)
Comment (bluebill1049) [reply src/useForm.ts]: It’s still wrong. I need to use get method

File: src/useForm.ts (hunk 3 above)
Comment (bluebill1049) [reply src/useForm.ts]: Shit that a mistake by merge

**Comment by bluebill1049** [comment]
Thanks @kotarella1110

File: src/useForm.ts (hunk 2 above)
Comment (bluebill1049) [reply src/useForm.ts]: oh man i just release so many things got wrongly merged. thanks a lot for pick this up!

**Pull request merged**